src/routes/rooms.ts

```diff
@@ -49,10 +49,18 @@ router.post(
   asyncHandler(async (req, res) => {
     const body: CreateRoom = req.body;
 
-    if (!body || Array.from(body.options).length < 1) {
-      res.status(HttpStatus.UNPROCESSABLE_ENTITY);
-      res.send();
-      return;
+    if (!body) {
+      throw new UnprocessableEntityError(
+        'ROOM_CREATE_EMPTY_DATA',
+        'Empty body sent',
+      );
+    }
+
+    if (!body.options || Array.from(body.options).length < 1) {
```

Comment:
nit: can we move this expression to a variable that describes what you are trying to check? this is kind of hard to understand.

Reply:
yup! I started by that but then I thought it wasn't that hard to understand.

Is this better?
```js
const emptyOptions = !body.options || Array.from(body.options).length < 1
```
Just keep in mind that we have a model/entity that is called `Option`, I want to change its name because it can be missleading

Reply:
what about ```areOptionsEmpty``` and flip the condition? negations in boolean names are always confusing for me 😨 (but it's probably my fault)

Reply:
you are not the only one :) I like `hasOptions` better, although I will need to negate the if condition, are u ok with that 😄 ? 
```js
const hasOptions = body.options && Array.from(body.options).length > 0;
if (!hasOptions) { ... }
```

Reply:
Yes!